substrafl_examples/go_further/run_diabetes_substrafl.py

```diff
@@ -0,0 +1,562 @@
+#
+# Next, we need to define the asset directory. You should have already downloaded the assets folder as stated above.
+#
+# A dataset represents the data in Substra. It contains some metadata and an *opener*, a script used to load the
+# data from files into memory. You can find more details about datasets
+# in the :ref:`API reference<documentation/references/sdk_schemas:DatasetSpec>`.
+
+from substra.sdk.schemas import DataSampleSpec
+from substra.sdk.schemas import DatasetSpec
+from substra.sdk.schemas import Permissions
+
+
+assets_directory = pathlib.Path.cwd() / "diabetes_substrafl_assets"
+assert assets_directory.is_dir(), """Did not find the asset directory,
+a directory called 'assets' is expected in the same location as this file"""
+
+permissions_dataset = Permissions(public=False, authorized_ids=DATA_PROVIDER_ORGS_ID)
+
+dataset = DatasetSpec(
+    name=f"Diabetes dataset",
+    type="csv",
+    data_opener=assets_directory / "dataset" / "diabetes_substrafl_opener.py",
+    description=data_path / "description.md",
+    permissions=permissions_dataset,
+    logs_permission=permissions_dataset,
+)
+
+# We register the dataset for each of the organizations
+dataset_keys = {client_id: clients[client_id].add_dataset(dataset) for client_id in DATA_PROVIDER_ORGS_ID}
+
+for client_id, key in dataset_keys.items():
+    print(f"Dataset key for {client_id}: {key}")
+
+
+# %%
+# The dataset object itself is an empty shell. Data samples are needed in order to add actual data.
+# A data sample contains subfolders containing a single data file like a CSV and the key identifying
+# the dataset it is linked to.
+
+datasample_keys = {
+    org_id: clients[org_id].add_data_sample(
+        DataSampleSpec(
+            data_manager_keys=[dataset_keys[org_id]],
+            test_only=False,
+            path=data_path / f"org_{i + 1}",
+        ),
+        local=True,
+    )
+    for i, org_id in enumerate(DATA_PROVIDER_ORGS_ID)
+}
+
+# %%
+# The flexibility of the ComputePlanBuilder class
+# ===============================================
+#
+# This example aims at explaining how to use the :ref:`substrafl_doc/api/compute_plan_builder:Compute Plan Builder`
+# class, and how to use the full power of the flexibility it provides.
+#
+# Before starting, we need to have in mind that a federated computation can be represented as a graph of tasks.
+# Some of these tasks needs data to be executed (training tasks) and others are here to aggregate local results
+# (aggregation tasks).
+#
+# Substra does not store an explicit definition of this graph; instead, it gives the user full flexibility to define
+# the compute plan (or computation graph) they need, by linking a task to its parents.
+#
+# To be able to create this graph of computation, SubstraFL provides the ``Node`` abstraction. A ``Node`` is used to
+# attached an organization (aka a Client) to a certain type of task depending on the computation we expect to happen on
+# this organization.
```

Comment:
```suggestion
# To create this graph of computations, SubstraFL provides the ``Node`` abstraction. A ``Node`` 
# assigns to an organization (aka a Client) tasks of a given type. The type of the ``Node`` depends on the type of tasks 
# we want to run on this organization (training or aggregation tasks).
```

Reply:
I find Sarah version easier to understand